src/main/java/use_cases/add_task/AddTaskUseCase.java

```diff
@@ -48,8 +48,9 @@ public AddTaskUseCase(AddTaskOutputBoundary presenter,
     public void addTask(String taskName, String taskDescription, int parentId,
                         int curriculumId) {
 
-        Task newTask = this.taskFactory.create(taskName, taskDescription);
-        TaskTree newTaskTree = this.taskTreeFactory.create();
+        TaskTree newTaskTree = this.taskTreeFactory.createWithTask(taskName, taskDescription);
+        Task newTask = newTaskTree.getTask();
```

Comment:
You have the correct name of (taskName, taskDescription), where you get to populate the tasktree. Good use of clean architecture as well, once I get Oswin's Ui your addTask feature should also work with my tree.